lib/internal/process/pre_execution.js

```diff
@@ -438,6 +440,47 @@ function setupInspectorHooks() {
   }
 }
 
+function setupNetworkInspection() {
+  if (!internalBinding('inspector').isEnabled() || !getOptionValue('--experimental-network-inspection')) {
+    return;
+  }
+  const dc = require('diagnostics_channel');
```

Comment:
These should probably done as a callback when the Network domain is enabled by the frontend/via Node.js's APIs?

Reply:
Speaking of which, we should probably also unsubscribe when the domain is disabled e.g. when the frontend is detached. I am not sure if there's already some better place to do the callback, but if not, maybe just throw some hacky logic into `dispatchProtocolMessage` and `sendMessageToFrontend` overrides to do the toggling?

Reply:
Actually making this enabled only by triggers might make it possible to be flagless without breaking anyone, because then it's a no-op when no one enables the network domain. I don't think many people use devtools://devtools/bundled/inspector.html to debug Node.js anyway? So we probably won't be breaking anyone using existing popular frontends.

Reply:
Changed to enable/disable network tracking with diagnostics_channel on `NodeNetwork.enable/disable` commands. e53ebec501fdc6adfbb3cb0b911664862ab64892